src/test_workflow/integ_test/distribution_rpm.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import subprocess
+
+from test_workflow.integ_test.distribution import Distribution
+
+
+class DistributionRpm(Distribution):
+    def __init__(self, filename: str, version: str, work_dir: str) -> None:
+        super().__init__(filename, version, work_dir)
+
+    @property
+    def get_install_dir(self) -> str:
+        return os.path.join(os.sep, "usr", "share", self.filename)
+
+    @property
+    def get_config_dir(self) -> str:
+        return os.path.join(os.sep, "etc", self.filename)
+
+    def install_distribution(self, bundle_name: str) -> None:
+        logging.info(f"Installing {bundle_name} in {self.get_install_dir}")
+        logging.info("rpm installation requires sudo, script will exit if current user does not have sudo access")
+        rpm_install_cmd = " ".join(
+            [
+                'yum',
+                'remove',
+                '-y',
+                self.filename,
+                '&&',
+                'yum',
+                'install',
+                '-y',
+                bundle_name
+            ]
+        )
+        subprocess.check_call(rpm_install_cmd, cwd=self.work_dir, shell=True)
+
+    @property
```

Comment:
You could subclass this further and get rid of all the maps. So `DistributionTar` becomes `DistributionOpenSearchTar` and ``DistributionOpenSearchDashboardsTar`. Then all these classes have different `start_cmd` properties that just returns the right value. You could get rid of `filename` altogether then. 

The other option is to keep `filename` and generate everything from it. So `start_cmd` is `f"systemctl start {filename}"` (or some transform of filename)

Reply:
The problem is in `tar` we can not generate everything as the start command for os and osd are completely different without any patterns.

My confusion is would we go this further to split classes this small, just to not use a map?

Thanks.

Reply:
After some further checking I think I will keep it this way.
With 90% of the code to be exactly the same and only 10% differs, it is hard for me to think separating to more subclasses would benefit the code.

Tho I would definitely get rid of one map on rpm, as rpm is very similar between os and osd, not tar.

Thanks.